Source/Blazorise/Components/Toast/Toast.razor.cs

```diff
@@ -70,12 +70,15 @@ public override async Task SetParametersAsync( ParameterView parameters )
     {
         if ( parameters.TryGetValue<bool>( nameof( Visible ), out var paramVisible ) && state.Visible != paramVisible )
         {
-            if ( paramVisible && await IsSafeToOpen() )
+            parameters.TryGetValue<Func<ToastOpeningEventArgs, Task>>( nameof( Opening ), out var paramOpening );
+            parameters.TryGetValue<Func<ToastClosingEventArgs, Task>>( nameof( Closing ), out var paramClosing );
+
+            if ( paramVisible && await IsSafeToOpen( paramOpening ) )
             {
                 await base.SetParametersAsync( parameters );
                 await SetVisibleState( true );
             }
```

Comment:
This makes it so `Opening` and `Closing` is called before  `base.SetParametersAsync( parameters ); `could this have any unintended side effects?

Reply:
What if it's not evaluated to true?
`await base.SetParametersAsync( parameters );` will not be called? parameters will not be updated?

Reply:
> What if it's not evaluated to true? `await base.SetParametersAsync( parameters );` will not be called? parameters will not be updated?

Hmm, we're calling `await base.SetParametersAsync( parameters );` in other `if` statements. But, yeah there might be a situation when it would not called. Let me refine it,

```cs
if ( parameters.TryGetValue<bool>( nameof( Visible ), out var paramVisible ) && state.Visible != paramVisible )
{
    parameters.TryGetValue<Func<ToastOpeningEventArgs, Task>>( nameof( Opening ), out var paramOpening );
    parameters.TryGetValue<Func<ToastClosingEventArgs, Task>>( nameof( Closing ), out var paramClosing );

    if ( paramVisible && await IsSafeToOpen( paramOpening ) )
    {
        await base.SetParametersAsync( parameters );
        await SetVisibleState( true );
    }
    else if ( !paramVisible && await IsSafeToClose( paramClosing ) )
    {
        await base.SetParametersAsync( parameters );
        await SetVisibleState( false );
    }

    if ( Rendered )
        await InvokeAsync( StateHasChanged );
}
else
{
    await base.SetParametersAsync( parameters );
}
```

Reply:
Ok, right, now, what about the order? The callbacks being called before ? Any concerns?

Reply:
> This makes it so `Opening` and `Closing` is called before `base.SetParametersAsync( parameters ); `could this have any unintended side effects?

I'm not sure, TBH. But considering that we always had the same logic in other places (like Modal), I would assume that it is fine unless we get it reported.